pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/FetchContext.java

```diff
@@ -83,4 +83,8 @@ public String getSegmentName() {
   public Map<String, List<ColumnIndexType>> getColumnToIndexList() {
     return _columnToIndexList;
   }
+
+  public boolean isEmpty() {
```

Comment:
didn't follow why we need this as a method

Reply:
as I'd like define the methods in FetchPlanner to always return non-null FetchContext, thus caller uses this method instead of `!=null`.